jmix-core/core/src/main/java/io/jmix/core/metamodel/datatype/impl/DoubleDatatype.java

```diff
@@ -82,10 +90,39 @@ public Double parse(String value, Locale locale) throws ParseException {
         }
 
         DecimalFormatSymbols formatSymbols = formatStrings.getFormatSymbols();
-        NumberFormat format = new DecimalFormat(formatStrings.getDoubleFormat(), formatSymbols);
+        DecimalFormat format = new DecimalFormat(formatStrings.getDoubleFormat(), formatSymbols);
+        format.setParseBigDecimal(true);
         return parse(value, format).doubleValue();
     }
 
+    protected Number parse(String value, NumberFormat format) throws ParseException {
+        BigDecimal result = (BigDecimal) super.parse(value, format);
+        if (coreProperties.isDecimalValueRoundByFormat()) {
+            int maximumFractionDigits = format.getMaximumFractionDigits();
+            RoundingMode roundingMode = format.getRoundingMode();
+            result = result.setScale(maximumFractionDigits, roundingMode);
+        }
+
+        if (!isInDoubleRange(result)) {
+            throw new ParseException(String.format("Double range exceeded: \"%s\"", value), 0);
```

Comment:
I'm not sure that "exceeded" is a proper word if we speak about negative values. Maybe we can say something like "The value is out of Double datatype range"? The same for Float.

Reply:
Done